api/storageprovisioner/provisioner.go

```diff
@@ -120,6 +120,12 @@ func (st *State) WatchVolumeAttachments(scope names.Tag) (watcher.MachineStorage
 	return st.watchAttachments("WatchVolumeAttachments", scope, apiwatcher.NewVolumeAttachmentsWatcher)
 }
 
+// WatchVolumeAttachmentPlans watches for changes to volume attachments
+// scoped to the entity with the tag passed to NewState.
+func (st *State) WatchVolumeAttachmentPlans(scope names.Tag) (watcher.MachineStorageIdsWatcher, error) {
```

Comment:
This and the other additions to the package could use tests. There are precedents in provisioner_test.go.